api/src/test/java/org/openmrs/api/handler/PatientDataUnvoidHandlerTest.java

```diff
@@ -144,4 +149,36 @@ public void handle_shouldNotUnvoidTheOrdersAndEncountersThatNeverGotVoidedWithTh
 		Assert.assertNotNull(testOrder.getVoidReason());
 		
 	}
+	
+	/**
+	 * @verifies unvoid the members associated with the patient
+	 * @see PatientDataUnvoidHandler#handle(Patient,User,Date,String)
+	 */
+	@Test
+	public void handle_shouldUnvoidMembersAssociatedWithThePatient() throws Exception {
+		executeDataSet(COHORT_XML);
+		CohortService cs = Context.getCohortService();
+		Cohort cohort = cs.getCohort(2);
+		Patient patient = Context.getPatientService().getPatient(7);
+		CohortMembership membership = new CohortMembership(patient);
+		cs.addMembershipToCohort(cohort, membership);
+		Assert.assertTrue(cohort.contains(patient));
+		
+		patient = Context.getPatientService().voidPatient(patient, "Void Reason");
+		Assert.assertTrue(patient.getVoided());
+		
+		Assert.assertTrue(membership.getVoided());
+		Assert.assertNotNull(membership.getDateVoided());
+		Assert.assertNotNull(membership.getVoidedBy());
+		Assert.assertNotNull(membership.getVoidReason());
+		
+		patient.setVoided(false);
```

Comment:
You should get rid of line 175 above because it's not the job of your code to call that, the reality is that the AOP around the unvoid method does reset those fields, this could be the reason the test is troubling you.